### PR TITLE
[jsdom] Use `base.d.ts` file

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
	- <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51642#issuecomment-800566999>
	- ~~<https://github.com/tc39/ecma262/pull/2089>~~ (extracted into its own PR: <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52022>)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

/cc @sandersn

### CODE REVIEW

**Comment by sandersn** [comment]
Quick question: why add a ts4.0/ version redirect as well? I don't see an explanation in the PR description.

**Comment by ExE-Boss** [comment]
@sandersn That’s because this PR also adds new built‑ins to `types/jsdom/index.d.ts`, which are included automatically when running on **Node 15**

**Comment by sandersn** [comment]
@ExE-Boss looks like the other owners aren't going to review this. I'll do it, but can you separate out the ts4.0 change into a separate PR? Having a structural PR and a semantic PR would help me review more easily.

**Review by RandinAk — APPROVED** [review]

File: types/jsdom/index.d.ts [hunk 1]
```diff
@@ -6,9 +6,10 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference path="ts3.5/index.d.ts"/>
+/// <reference path="./base.d.ts"/>
```
Comment (sandersn) [inline types/jsdom/index.d.ts:9]: Why does the main index need to reference both base and ts3.5's index? I thought the point of the change was to keep the common core in base.

**Review by sandersn — CHANGES_REQUESTED** [review]
Question about the two `/// <reference` in index.d.ts

**Review by sandersn — APPROVED** [review]

**Comment by ExE-Boss** [comment]
Ready to merge

**Pull request merged**